src/components/cylc/workflow/Toolbar.vue

```diff
@@ -81,23 +81,30 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
       >
         <template v-slot:activator="{ on }">
           <a class="add-view" v-on="on">
-            {{ $t('Toolbar.addView') }} <v-icon color="#5995EB">{{ svgPaths.add }}</v-icon>
+            <v-icon class="icon" color="#5995EB">{{ svgPaths.add }}</v-icon>
+            <span class="label">
+              {{ $t('Toolbar.addView') }}
+            </span>
           </a>
         </template>
         <v-list class="pa-0">
           <v-list-item
-            class="py-0 px-8 ma-0"
-            @click="$listeners['add-tree']"
-            id="toolbar-add-tree-view"
+            v-bind:class="[
+              'py-0',
+              'px-8',
+              'ma-0',
+              'c-add-view',
+              `c-add-view-${ view.name }`
+            ]"
+            :id="`toolbar-add-${ view.name }-view`"
+            @click="$listeners['add']"
+            v-for="view in views"
+            v-bind:key="view.title"
```

Comment:
I think this could be simpler.

If I understood it correctly, we have a list in the Vue component with the views. For each view we iterate it and create a CSS class with the view name.

Then we fire an event on click. The method bound to the event will parse the CSS class and parse the view name. I think we should be able to call the listener method passing the view name. That eliminates the need for a dynamic css class, and the need to parse the CSS class to retrieve the view name.